application/src/test/java/org/opentripplanner/street/model/edge/StreetEdgeGeofencingTest.java

```diff
@@ -284,17 +272,184 @@ public void pickupFloatingVehiclesWhenStartedInNoDropOffZone() {
+      var states = runTraverse(Set.of(NETWORK_TIER), Collections.emptySet());
+
+      // Walking, unknown, Tier.
+      assertEquals(3, states.length);
+      final State walkState = states[0];
+      assertEquals(HAVE_RENTED, walkState.getVehicleRentalState());
+      assertEquals(WALK, walkState.currentMode());
+
+      final State unknownNetworkState = states[1];
+      assertEquals(RENTING_FLOATING, unknownNetworkState.getVehicleRentalState());
+      assertEquals(BICYCLE, unknownNetworkState.currentMode());
+      assertNull(unknownNetworkState.getVehicleRentalNetwork());
+
+      final State tierState = states[2];
+      assertEquals(RENTING_FLOATING, tierState.getVehicleRentalState());
+      assertEquals(BICYCLE, tierState.currentMode());
+      assertEquals(NETWORK_TIER, tierState.getVehicleRentalNetwork());
+    }
+
+    @Test
+    public void pickupFloatingVehiclesWhenStartedInNoDropOffZoneAllNetworksBanned() {
+      var states = runTraverse(Collections.emptySet(), Set.of(NETWORK_TIER, NETWORK_BIRD));
+
+      // Should only have a walking state. The unknown network state should only be
+      // generated if there are known network states.
+      assertEquals(1, states.length);
+      final State walkState = states[0];
+      assertEquals(HAVE_RENTED, walkState.getVehicleRentalState());
+      assertEquals(WALK, walkState.currentMode());
+    }
+
+    @Test
+    public void pickupFloatingVehiclesWhenStartedInNoDropOffZoneSomeNetworksBanned() {
+      var states = runTraverse(Collections.emptySet(), Set.of(NETWORK_BIRD));
+
+      // Walking state, unknown network state, known network state for Tier (which wasn't banned)
+      assertEquals(3, states.length);
+
+      final State walkState = states[0];
+      assertEquals(HAVE_RENTED, walkState.getVehicleRentalState());
+      assertEquals(WALK, walkState.currentMode());
+
+      final State unknownNetworkState = states[1];
+      assertEquals(RENTING_FLOATING, unknownNetworkState.getVehicleRentalState());
+      assertEquals(BICYCLE, unknownNetworkState.currentMode());
+      assertNull(unknownNetworkState.getVehicleRentalNetwork());
+
+      final State tierState = states[2];
+      assertEquals(RENTING_FLOATING, tierState.getVehicleRentalState());
+      assertEquals(BICYCLE, tierState.currentMode());
+      assertEquals(NETWORK_TIER, tierState.getVehicleRentalNetwork());
+    }
+
+    private State[] runTraverse(Set<String> allowedNetworks, Set<String> bannedNetworks) {
+      var req = makeArriveByRequest(allowedNetworks, bannedNetworks);
+
+      V2.addRentalRestriction(NO_DROP_OFF_TIER);
+      V2.addRentalRestriction(noDropOffRestriction(NETWORK_BIRD));
+
+      var haveRentedState = makeHaveRentedState(V2, req);
+
+      var edge = streetEdge(V1, V2);
+
+      return edge.traverse(haveRentedState);
+    }
+
+    private static State makeHaveRentedState(Vertex vertex, StreetSearchRequest req) {
```

Comment:
I think we mostly use "create" instead of "make" in these types of method names. However, I don't think that is written down in naming conventions and I was surprised to see how often we use "make" in these types of method names. I can bring this naming convention thing up in a dev meeting.